lib/api/2.0/templates.js

```diff
@@ -0,0 +1,89 @@
+// Copyright 2016, EMC Inc.
+
+'use strict';
+
+var injector = require('../../../index').injector;
+var swagger = injector.get('Http.Services.Swagger');
+var controller = swagger.controller;
+var templates = injector.get('Templates');
+var taskProtocol = injector.get('Protocol.Task');
+var Errors = injector.get('Errors');
+var lookups = injector.get('Services.Lookup');
+var waterline = injector.get('Services.Waterline');
+var workflowApiService = injector.get('Http.Services.Api.Workflows');
+var _ = injector.get('_');
+
+// GET /templates/metadata
+var templatesMetaGet = controller(function() {
+    return templates.getAll();
+});
+
+// GET /templates/metadata/:name
+var templatesMetaGetByName = controller(function(req) {
+    return templates.getName(req.swagger.params.name.value,
+                             req.swagger.query.scope);
+});
+
+// GET /templates/:name
+var templatesGetByName = controller(function(req, res) {
+    return lookups.ipAddressToMacAddress(res.locals.ipAddress)
+    .then(function(mac) {
```

Comment:
@yyscamper `ipAddressToMacAddress` calls `lookups.findOneByTerm` which will throw a `Errors.NotFoundError` if the lookup record is not found